### PR TITLE
Encrypt migration export files with AES-256-GCM

### PR DESCRIPTION
**Pull request by MatthieuTexier**
## Problem

The `.columba` export file contains private keys, messages, contacts, and all user data in **plaintext**. Since this file transits between devices (email, cloud storage, messaging apps…), it is the most exposed attack vector.

## Solution

All `.columba` exports are now encrypted with **AES-256-GCM** using a user-chosen password, with key derivation via **PBKDF2-HMAC-SHA256** (600,000 iterations).

**Encrypted file format:**
```
[1 byte:  version 0x02]
[16 bytes: PBKDF2 salt]
[12 bytes: AES-GCM IV/nonce]
[N bytes:  AES-256-GCM ciphertext + 16-byte auth tag]
```

## Changes

- **`MigrationCrypto.kt`** (new): AES-256-GCM encryption/decryption utility with PBKDF2 key derivation, format detection, and custom exceptions
- **`MigrationExporter.kt`**: `exportData()` now takes a `password` parameter and encrypts the ZIP after creation
- **`MigrationImporter.kt`**: Auto-detects encrypted vs legacy files (ZIP magic bytes `0x50 0x4B` vs version byte `0x02`), decrypts when needed, handles wrong password and missing password errors
- **`MigrationViewModel.kt`**: New `PasswordRequired` and `WrongPassword` UI states, password flow for export and import
- **`MigrationScreen.kt`**: Password dialog with confirmation for export, single-field prompt for import, validation (min 8 chars, mismatch check), wrong password error display

## Backward compatibility

Unencrypted `.columba` files from older versions import **without any password prompt** — detection is automatic via the first two bytes of the file.

## Tests

**92 migration-related tests passing**, including:
- 16 unit tests for `MigrationCrypto` (round-trip, wrong password, unicode, edge cases)
- 8 Robolectric tests for `MigrationImporter` encryption (format detection, encrypted preview, wrong/missing password)
- 16 Compose UI tests for `PasswordDialog` (validation, modes, callbacks, integration with screen states)
- 16 ViewModel tests updated for new signatures
- All 36 pre-existing migration tests still passing

### CODE REVIEW

**Comment by torlando-tech** [comment]
@greptileai review

**Comment by torlando-tech** [comment]
@greptileai

File: app/src/main/java/com/lxmf/messenger/ui/screens/MigrationScreen.kt [hunk 1]
```diff
@@ -732,3 +784,120 @@ private fun NotificationPermissionDialog(
...
+    description: String,
+    isConfirmMode: Boolean,
+    isWrongPassword: Boolean,
+    onConfirm: (String) -> Unit,
+    onDismiss: () -> Unit,
+) {
+    var password by remember { mutableStateOf("") }
+    var confirmPassword by remember { mutableStateOf("") }
+    var passwordVisible by remember { mutableStateOf(false) }
+    var errorMessage by remember { mutableStateOf<String?>(if (isWrongPassword) "Incorrect password" else null) }
```
[the remark this answers is not in the document]
Comment (torlando-tech) [reply app/src/main/java/com/lxmf/messenger/ui/screens/MigrationScreen.kt:810]: false, this works fine

File: app/src/main/java/com/lxmf/messenger/ui/screens/MigrationScreen.kt [hunk 2]
```diff
@@ -732,3 +784,120 @@ private fun NotificationPermissionDialog(
...
+
+    val minLength = com.lxmf.messenger.migration.MigrationCrypto.MIN_PASSWORD_LENGTH
+
+    fun validate(): Boolean {
+        if (password.length < minLength) {
+            errorMessage = "Password must be at least $minLength characters"
+            return false
+        }
```
[the remark this answers is not in the document]
Comment (torlando-tech) [reply app/src/main/java/com/lxmf/messenger/ui/screens/MigrationScreen.kt:818]: edge case i'm not concerned with
